git-subhistory.sh

```diff
@@ -267,18 +273,27 @@ subhistory_assimilate () {
 		revs_to_rewrite=ASSIMILATE_HEAD
 	fi
 
-	git filter-branch \
+        commit_count=$(git rev-list --count $revs_to_rewrite)
+
+        if test $commit_count -ne 0
+        then
+	    git filter-branch \
 		--original subhistory-tmp/filter-branch-backup \
 		--parent-filter "$parent_filter" \
 		--index-filter "$index_filter"  \
 		-- $revs_to_rewrite \
-	2>&1 | say_stdin || exit $?
-
-	say
-	say "Assimilated $assimilatee into $(
-		git symbolic-ref --short HEAD -q \
-		|| echo "detached HEAD ($(git rev-parse --short HEAD))"
-	) under $path_to_sub as ASSIMILATE_HEAD"
+	        2>&1 | say_stdin || exit $?
+
+	    say
+	    say "Assimilated $assimilatee into $(
+		    git symbolic-ref --short HEAD -q \
+		    || echo "detached HEAD ($(git rev-parse --short HEAD))"
+	    ) under $path_to_sub as ASSIMILATE_HEAD"
+        else
+            say
+            say "Already completely assimilated - nothing to do"
```

Comment:
```
* Handle an empty merge by detecting that there is nothing to assimilate
  and cleanly returning (instead of returning an error from "git
  filter-branch" that there's nothing to do).
```

whoops good point! This should be easy to add a test case for